src/Kiota.Builder/Writers/Php/CodeMethodWriter.cs

```diff
@@ -70,22 +70,13 @@ public override void  WriteCodeElement(CodeMethod codeElement, LanguageWriter wr
         private static void WriteConstructorBody(CodeClass parentClass, CodeMethod currentMethod, LanguageWriter writer, bool inherits) {
             if(inherits)
                 writer.WriteLine("parent::__construct();");
-            foreach(var propWithDefault in parentClass.GetPropertiesOfKind(
-                                                                            CodePropertyKind.BackingStore,
-                                                                            CodePropertyKind.RequestBuilder,
-                                                                            CodePropertyKind.UrlTemplate,
-                                                                            CodePropertyKind.PathParameters)
+            foreach(var propWithDefault in parentClass.Properties
                                             .Where(x => !string.IsNullOrEmpty(x.DefaultValue))
                                             .OrderByDescending(x => x.Kind)
                                             .ThenBy(x => x.Name))
             {
                 var isPathSegment = propWithDefault.IsOfKind(CodePropertyKind.PathParameters);
-                writer.WriteLine($"$this->{propWithDefault.NamePrefix}{propWithDefault.Name.ToFirstCharacterLowerCase()} = {(isPathSegment ? "[]" :propWithDefault.DefaultValue.ReplaceDoubleQuoteWithSingleQuote())};");
```

Comment:
the reason why the additional data was done in a second phase is because additional data can be stored in the backing store and would need it to be initialized first. And ordering by kind only is not enough.
Although ordering could work with a custom comparer.

Reply:
Does this mean that custom properties and additional properties should be initialised via setters?
Not very familiar with how the backing store works...

Reply:
correct

Reply:
updated